eZ/Publish/Core/FieldType/Image/ImageStorage.php

```diff
@@ -113,6 +113,10 @@ public function storeFieldData(VersionInfo $versionInfo, Field $field, array $co
                 return false;
             }
 
+            if (!isset($field->value->data['id'])) {
```

Comment:
I don't know it that well, but AFAICS empty image has still `ValueObject`, empty though. Also there's "empty"-ish XML for empty image (`ImageConverter::createEmptyLegacyXml`) which is the reason why the code you've linked above looks like this.

I need to look deeper into this to see what happens here :)